### PR TITLE
Revolver Rework Attempt 2- With new sounds and better testing. Uplink changes.

### PR DESCRIPTION
**Pull request by Pariah919**
Adds 15mm to uplink.
Makes revolvers break open to be able to reload them.
Better messages than last time for player clairity.
Tested more. Apologies.
Better checks.
https://streamable.com/tf08cr
image

### CODE REVIEW

**Comment by InfinitelyThinRectangles** [comment]
I'm gonna repost this, because it's important.

---------------------------------------

Hi sorry, just a quick one.

Look I hate to be the fun police and take a shit on people's work but uh, nobody asked for this, nobody wants this, and the weapons this nerfs by adding an extra micromanagement step that no other weapon in the game has in the same way aren't anywhere near actually good enough or common enough, either in use or just in existence to warrant this added pain in the balls trying to use them.

I understand, I really do, the exhilaration of finding a cool quick thing you can do and push to live that came from a learning experience, but this isn't a very good change.

Can it NOT come back?

Please just let people have fun.

---------------------------------------

As for this:

its a cool mechanic. gives revolvers a bit more weight when you use them. if it bothers you that much, don't pick one. there's currently no real downside to picking the revolver over the pistol besides having a few more shots on the pistol, plus this barely effects your time to reload revolvers if you can click.

--------------------------------------------


No. No, no, no, no, no, _no, no. No._ "It's a cool mechanic" is a meaningless sentence, it does nothing except include a single black line 3 pixels long on a sprite and involves extra clicks.

"gives revolvers a bit more weight when you use them"

what2

_**What does this even mean?!**_

"if it bothers you that much, don't pick one" is possibly the single worst excuse for fucking something up because you want to with no regard to the fact that what you're trying to do here effects everyone else who might have reasons for using something, I don't think I've ever seen such wanton selfishness.

 "there's currently no real downside" - "besides having a few more shots on the pistol," I mean you seem to have rebuffed your own argument there so I have no further need to comment.
 
 "plus this barely effects your time to reload revolvers if you can click."  Absolutely 1 billion percent not my point. You've decided with no community consultation to the people who actually use this thing that you're going to modify an inferior choice of weapon that _doesn't need it_ with a nerf seen nowhere else in the game, for absolutely no particularly good reason that you've articulated here other than "well I like it and feel entitled to alter the gear everyone else has to use because I feel like it."
 
 Rifles don't have to have their bolt cycled when you load a new magazine from empty, semi autos don't need their slide racked when you reload them from empty, why do inferior weapons that are in no way shape or form whatsoever an issue being nerfed via increasing the complexity of their manual of arms?

If for some reason there's an absolutely **BURNING** need to nerf the revolvers that barely exist and are mediocre to crap options and for some insane reason the preferred manner to do this is by making them harder to reload, _just give it an action timer_ instead of actively making something fiddlier.

There is NEVER an excuse for weaponising inconvenience to make something less viable in game design. Because that's what SS13 is, a _game,_ that is _played,_ and the _play_ of such _game_ needs to be taken into account when changes are made. This whole concept **isn't required from a gameplay standpoint** and just damages something that's absolutely fine as-is, it's change for changes sake, it doesn't solve any of the issues that don't exist to be solved anyway and it adds absolutely nothing except irritation to the experience of using or fighting someone with a revolver.

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 1]
```diff
@@ -11,12 +11,28 @@
...
 	ammo_type = /obj/item/ammo_casing/pistol/magnum
 	var/chamber_offset = 0 //how many empty chambers in the cylinder until you hit a round
+	fire_sound = 'sound/weapons/gunshot/revolver_medium.ogg'
 	mag_insert_sound = 'sound/weapons/guns/interaction/rev_magin.ogg'
 	mag_remove_sound = 'sound/weapons/guns/interaction/rev_magout.ogg'
 	accuracy = 2
 	accuracy_power = 8
 	one_hand_penalty = 2
 	bulk = 3
+	var/broke_open = FALSE // Revovlers break open to reveal a cylinder.
```
Comment (Kaksisilma) [inline code/modules/projectiles/guns/projectile/revolver.dm]:
```suggestion
	var/broke_open = FALSE // Revolvers break open to reveal a cylinder.
```

**Review by Kaksisilma — COMMENTED** [review]
minor grammar nitpick

**Comment by Mannybrado** [comment]
absolutely.
> I'm gonna repost this, because it's important.
> 
> Hi sorry, just a quick one.
> 
> Look I hate to be the fun police and take a shit on people's work but uh, nobody asked for this, nobody wants this, and the weapons this nerfs by adding an extra micromanagement step that no other weapon in the game has in the same way aren't anywhere near actually good enough or common enough, either in use or just in existence to warrant this added pain in the balls trying to use them.
> 
> I understand, I really do, the exhilaration of finding a cool quick thing you can do and push to live that came from a learning experience, but this isn't a very good change.
> 
> Can it NOT come back?
> 
> Please just let people have fun.
> 
> As for this:
> 
> its a cool mechanic. gives revolvers a bit more weight when you use them. if it bothers you that much, don't pick one. there's currently no real downside to picking the revolver over the pistol besides having a few more shots on the pistol, plus this barely effects your time to reload revolvers if you can click.
> 
> No. No, no, no, no, no, _no, no. No._ "It's a cool mechanic" is a meaningless sentence, it does nothing except include a single black line 3 pixels long on a sprite and involves extra clicks.
> 
> "gives revolvers a bit more weight when you use them"
> 
> what2
> 
> _**What does this even mean?!**_
> 
> "if it bothers you that much, don't pick one" is possibly the single worst excuse for fucking something up because you want to with no regard to the fact that what you're trying to do here effects everyone else who might have reasons for using something, I don't think I've ever seen such wanton selfishness.
> 
> "there's currently no real downside" - "besides having a few more shots on the pistol," I mean you seem to have rebuffed your own argument there so I have no further need to comment.
> 
> "plus this barely effects your time to reload revolvers if you can click." Absolutely 1 billion percent not my point. You've decided with no community consultation to the people who actually use this thing that you're going to modify an inferior choice of weapon that _doesn't need it_ with a nerf seen nowhere else in the game, for absolutely no particularly good reason that you've articulated here other than "well I like it and feel entitled to alter the gear everyone else has to use because I feel like it."
> 
> Rifles don't have to have their bolt cycled when you load a new magazine from empty, semi autos don't need their slide racked when you reload them from empty, why do inferior weapons that are in no way shape or form whatsoever an issue being nerfed via increasing the complexity of their manual of arms?
> 
> If for some reason there's an absolutely **BURNING** need to nerf the revolvers that barely exist and are mediocre to crap options and for some insane reason the preferred manner to do this is by making them harder to reload, _just give it an action timer_ instead of actively making something fiddlier.
> 
> There is NEVER an excuse for weaponising inconvenience to make something less viable in game design. Because that's what SS13 is, a _game,_ that is _played,_ and the _play_ of such _game_ needs to be taken into account when changes are made. This whole concept **isn't required from a gameplay standpoint** and just damages something that's absolutely fine as-is, it's change for changes sake, it doesn't solve any of the issues that don't exist to be solved anyway and it adds absolutely nothing except irritation to the experience of using or fighting someone with a revolver.

I think you're overreacting. Let's walk through the steps to reload a revolver.
First, of course, fire all of your shots.
Press Z. This ejects all of your casings and opens the cylinder.
Get your speed loader if you haven't already.
Put it in the gun.
Press Z to close the cylinder.(the only new action with this PR)
This is barely a nerf. If you do it fast enough it is literally maybe less than a fraction of a second slower. This is _barely_ a nerf.
Personally? I believe you should have to rack the chamber on empty semi-autos. For bolt actions, they're so rarely used I don't think changing them really matters, but yes, having a proper bolting action would be fun. 
Guns are quite powerful on this codebase. Adding some mechanical tedium increases the chance for human error and promotes actually getting good at using a specific type of weapon. This is what I mean by "weight". If you fumble around with it that much, I don't really know what to say.
Also, adding an action timer would make it actively harder to reload than what this PR does. Having a chance to jam but you have a couple more shots is a significantly worse to having no jam chance. RNG in combat will fuck you more than anything else will.
Finally, "community consolation"? Lol.

**Comment by ProbablyCarl** [comment]
Going to go ahead and test merge this.

**Comment by pvtSPEDwarrior** [comment]
> I'm gonna repost this, because it's important.
> 
> Hi sorry, just a quick one.
> 
> Look I hate to be the fun police and take a shit on people's work but uh, nobody asked for this, nobody wants this, and the weapons this nerfs by adding an extra micromanagement step that no other weapon in the game has in the same way aren't anywhere near actually good enough or common enough, either in use or just in existence to warrant this added pain in the balls trying to use them.
> 
> I understand, I really do, the exhilaration of finding a cool quick thing you can do and push to live that came from a learning experience, but this isn't a very good change.
> 
> Can it NOT come back?
> 
> Please just let people have fun.
> 
> As for this:
> 
> its a cool mechanic. gives revolvers a bit more weight when you use them. if it bothers you that much, don't pick one. there's currently no real downside to picking the revolver over the pistol besides having a few more shots on the pistol, plus this barely effects your time to reload revolvers if you can click.
> 
> No. No, no, no, no, no, _no, no. No._ "It's a cool mechanic" is a meaningless sentence, it does nothing except include a single black line 3 pixels long on a sprite and involves extra clicks.
> 
> "gives revolvers a bit more weight when you use them"
> 
> what2
> 
> _**What does this even mean?!**_
> 
> "if it bothers you that much, don't pick one" is possibly the single worst excuse for fucking something up because you want to with no regard to the fact that what you're trying to do here effects everyone else who might have reasons for using something, I don't think I've ever seen such wanton selfishness.
> 
> "there's currently no real downside" - "besides having a few more shots on the pistol," I mean you seem to have rebuffed your own argument there so I have no further need to comment.
> 
> "plus this barely effects your time to reload revolvers if you can click." Absolutely 1 billion percent not my point. You've decided with no community consultation to the people who actually use this thing that you're going to modify an inferior choice of weapon that _doesn't need it_ with a nerf seen nowhere else in the game, for absolutely no particularly good reason that you've articulated here other than "well I like it and feel entitled to alter the gear everyone else has to use because I feel like it."
> 
> Rifles don't have to have their bolt cycled when you load a new magazine from empty, semi autos don't need their slide racked when you reload them from empty, why do inferior weapons that are in no way shape or form whatsoever an issue being nerfed via increasing the complexity of their manual of arms?
> 
> If for some reason there's an absolutely **BURNING** need to nerf the revolvers that barely exist and are mediocre to crap options and for some insane reason the preferred manner to do this is by making them harder to reload, _just give it an action timer_ instead of actively making something fiddlier.
> 
> There is NEVER an excuse for weaponising inconvenience to make something less viable in game design. Because that's what SS13 is, a _game,_ that is _played,_ and the _play_ of such _game_ needs to be taken into account when changes are made. This whole concept **isn't required from a gameplay standpoint** and just damages something that's absolutely fine as-is, it's change for changes sake, it doesn't solve any of the issues that don't exist to be solved anyway and it adds absolutely nothing except irritation to the experience of using or fighting someone with a revolver.

L + Ratio

**Comment by Pariah919** [comment]
Not going to quote the entire thing to stop this threat from being huge, but-

>Look I hate to be the fun police and take a shit on people's work but uh, nobody asked for this, nobody wants this, and the weapons this nerfs by adding an extra micromanagement step that no other weapon in the game has in the same way aren't anywhere near actually good enough or common enough, either in use or just in existence to warrant this added pain in the balls trying to use them.
>
I don't mind you giving criticism on my work, nothing deflects from being able to give some even to things that are WIP (which this isn't.), but personally I prefer this to the old bay system because.. It feels better?
Weapon feel is the most important thing in ss13 to making guns feel better to use, if we wanted revolvers to just be fancy pistols with barely any difference, might as well make all of them deagles or something.

>nobody asked for this, nobody wants 
>
No offense, but I wanted it. So I made it, the PR and my argument for it being added to the game is that it makes revolvers feel overall better to use and a bit more unique, I think it'll be fine considering I've barely gotten any feedback, which means its not that noticeable in the first place, which means it went as about I expected.
This makes revolvers feel more unique in comparison to other weapon types, even if its something insanely minor. You need to press Z, what. Once more?

>No. No, no, no, no, no, no, no. No. "It's a cool mechanic" is a meaningless sentence, it does nothing except include a single black line 3 pixels long on a sprite and involves extra clicks.
>
i've nver said cool mechanic about this ever? I think its a decent mechanic, I like the sounds and the feel of closing and opening the revolver, its half the reason I used it in TGMC constnatly for a while.

>"if it bothers you that much, don't pick one" is possibly the single worst excuse for fucking something up because you want to with no regard to the fact that what you're trying to do here effects everyone else who might have reasons for using something, I don't think I've ever seen such wanton selfishness.
>
no offense but I've never said this, I think people will like this generally. As I've gotten a decent amount of support about adding it.

**Comment by ProbablyCarl** [comment]
We've had this test merged for a while. Pending being fully merged.

**Pull request merged**